### ISSUE TITLE
[Request for insight] result of layered colors

### ISSUE THREAD

**Issue by haxxxton** [Qix-/color#154]
I've been banging my head about this one for a while, and cant quite figure out why the resultant colors between mathematically calculated values (ie, those from `color`) and those inside a image editing program (in this case Sketch), are different.

The scenario i find myself is in this:
Our design team has created a color scale from 0% to 200%. 100% being the pure color (eg a blue `#007FEF`), they then overlay, in opacity steps of 10%, a light color (`#FFFFFF`) and a dark color (`#14181C`) moving away from this 100 in opposite directions. Leading to 0% being the pure light color, and 200% being the pure dark color. Below is an example of said spectrum.

![image](https://user-images.githubusercontent.com/743105/49358070-21eb9680-f71d-11e8-88ad-86bbb5a91312.png)

The resultant colors of this overlay technique are determined by using the color picker/dropper tool on the overlayed colors.

When trying to replicate this scale inside `color`, I am trying to make generation of the colors as easy as possible. I assumed that `.mix()` would be best solution for this, however, the resulting hex values are slightly off from the resulting hex colors in my image editing program.

```
const baseColor = Color('#007FEF');
const lightOverlay = Color('#FFFFFF');
const darkOverlay = Color('#14181C');
/*
mix code				generated output  expected output  matching
*/
baseColor.mix(lightOverlay, 1).hex();	// #FFFFFF	// #FFFFFF	// ===
baseColor.mix(lightOverlay, 0.9).hex();	// #E6F2FD	// #E6F2FD	// ===
baseColor.mix(lightOverlay, 0.8).hex();	// #CCE5FC	// #CCE5FB	// =/=
baseColor.mix(lightOverlay, 0.7).hex();	// #B3D9FA	// #B3D8FA	// =/=
baseColor.mix(lightOverlay, 0.6).hex();	// #99CCF9	// #99CBF8	// =/=
baseColor.mix(lightOverlay, 0.5).hex();	// #80BFF7	// #80BFF7	// ===
baseColor.mix(lightOverlay, 0.4).hex();	// #66B2F5	// #66B2F5	// ===
baseColor.mix(lightOverlay, 0.3).hex();	// #4DA5F4	// #4DA5F3	// =/=
baseColor.mix(lightOverlay, 0.2).hex();	// #3399F2	// #3398F2	// =/=
baseColor.mix(lightOverlay, 0.1).hex();	// #198CF1	// #1A8CF0	// =/=
baseColor.mix(darkOverlay, 0).hex();	// #007FEF	// #007FEF	// ===
baseColor.mix(darkOverlay, 0.1).hex();	// #0275DA	// #0274D9	// =/=
baseColor.mix(darkOverlay, 0.2).hex();	// #046AC5	// #046AC5	// ===
baseColor.mix(darkOverlay, 0.3).hex();	// #0660B0	// #065FAE	// =/=
baseColor.mix(darkOverlay, 0.4).hex();	// #08569B	// #08569A	// =/=
baseColor.mix(darkOverlay, 0.5).hex();	// #0A4C86	// #0A4B85	// =/=
baseColor.mix(darkOverlay, 0.6).hex();	// #0C4170	// #0C4070	// =/=
baseColor.mix(darkOverlay, 0.7).hex();	// #0E375B	// #0E365B	// =/=
baseColor.mix(darkOverlay, 0.8).hex();	// #102D46	// #102C45	// =/=
baseColor.mix(darkOverlay, 0.9).hex();	// #122231	// #122230	// =/=
baseColor.mix(darkOverlay, 1).hex();	// #14181C	// #14181C	// ===
```

Now admittedly, the difference between the two colors is so marginal (<1%) and visually I personally cant tell the difference, I have been tasked with seeing if there is a way to make things match perfectly. There are some base colors where this difference is a little more noticeable (~4% difference) at certain percentages of mixing, this is particularly the case for our bluegray base color `#3D4C5C`. For my own sanity I would prefer not to have to manually type out each hex color (we have ~10 base colors, and some step by 5% instead of 10% opacity).

**TLDR;**
Is there a blending/mixing function in `color` that will mirror exactly the visual output achieved by overlaying a transparent color over an opaque color in Sketch?

**Comment by Qix-** [comment]
Color overlaying isn't an exact thing, there are multiple ways to tackle the problem.

This could very well be a rounding discrepancy seeing as how they're so close. The colors are generally converted to floating point and then rounded off (either using `.floor`, `.ceil` or `.round`), each giving you a slightly different result in different cases.

**Comment by haxxxton** [comment]
@Qix- thanks so much for the speedy reply! Sounds like a great idea! i was worried that you were going to say something about needing to do some crazy color mode switching stuff to get it to work. Glad that `.mix` is at least the right way to do this :D
